processing/src/main/java/org/apache/druid/query/filter/TypedInFilter.java

```diff
@@ -222,12 +254,9 @@ public RangeSet<String> getDimensionRangeSet(String dimension)
       return null;
     }
     RangeSet<String> retSet = TreeRangeSet.create();
-    for (Object value : lazyMatchValues.get()) {
-      String valueEquivalent = NullHandling.nullToEmptyIfNeeded(Evals.asString(value));
+    for (Object value : sortedMatchValues.get()) {
+      String valueEquivalent = Evals.asString(value);
       if (valueEquivalent == null) {
-        // Case when SQL compatible null handling is enabled
-        // Range.singleton(null) is invalid, so use the fact that
```

Comment:
Most of this comment is still useful in explaining why we're doing `lessThan("")`.